packages/lodestar/src/chain/attestation.ts

```diff
@@ -113,13 +113,12 @@ export class AttestationProcessor implements IAttestationProcessor {
     const targetSlot = computeStartSlotAtEpoch(this.config, target.epoch);
     const ancestor = this.forkChoice.getAncestor(attestation.data.beaconBlockRoot as Uint8Array, targetSlot);
     assert.true(
-      ancestor && this.config.types.Root.equals(target.root, ancestor),
+      Boolean(ancestor && this.config.types.Root.equals(target.root, ancestor)),
       "FFG and LMD vote must be consistent with each other");
     const stateCtx = await this.db.stateCache.get(block.stateRoot);
-    assert.true(
-      !!stateCtx,
-      `Missing state context for attestation block with stateRoot ${toHexString(block.stateRoot)}`
-    );
+    if (stateCtx === null) {
+      throw Error(`Missing state context for attestation block with stateRoot ${toHexString(block.stateRoot)}`)
+    }
```

Comment:
Is it necessary to throw the error with `assert`? If not better, because tsc can see that in this branch `stateCtx` is defined

Reply:
Well it's not but I don't think there is a way to indicate tsc that this method will throw if param is false

Reply:
What do you mean with "if param is false"? Should I do any further changes to this file?

Reply:
well it would seem tsc doesn't check method implementation to see if it will throw so nope, we cannot do anything about it, I think